src/core/SpyFactory.js

```diff
@@ -1,45 +1,61 @@
 getJasmineRequireObj().SpyFactory = function(j$) {
 
-  function SpyFactory(getCustomStrategies) {
-    var self = this;
-
-    this.createSpy = function(name, originalFn) {
-      return j$.Spy(name, originalFn, getCustomStrategies());
-    };
-
-    this.createSpyObj = function(baseName, methodNames) {
-      var baseNameIsCollection = j$.isObject_(baseName) || j$.isArray_(baseName);
-
-      if (baseNameIsCollection && j$.util.isUndefined(methodNames)) {
-        methodNames = baseName;
-        baseName = 'unknown';
-      }
-
-      var obj = {};
-      var spiesWereSet = false;
-
-      if (j$.isArray_(methodNames)) {
-        for (var i = 0; i < methodNames.length; i++) {
-          obj[methodNames[i]] = self.createSpy(baseName + '.' + methodNames[i]);
-          spiesWereSet = true;
-        }
-      } else if (j$.isObject_(methodNames)) {
-        for (var key in methodNames) {
-          if (methodNames.hasOwnProperty(key)) {
-            obj[key] = self.createSpy(baseName + '.' + key);
-            obj[key].and.returnValue(methodNames[key]);
-            spiesWereSet = true;
-          }
-        }
-      }
-
-      if (!spiesWereSet) {
-        throw 'createSpyObj requires a non-empty array or object of method names to create spies for';
-      }
-
-      return obj;
-    };
-  }
-
-  return SpyFactory;
+    function SpyFactory(getCustomStrategies) {
+        var self = this;
+
+        this.createSpy = function(name, originalFn) {
+            return j$.Spy(name, originalFn, getCustomStrategies());
+        };
+
+        this.createSpyObj = function(baseName, methodNames, propertyNames) {
+            var baseNameOrMethodNamesAreCollections = j$.isObject_(baseName) || j$.isArray_(baseName) || j$.isObject_(propertyNames) || j$.isArray_(propertyNames);
+
+            if (baseNameOrMethodNamesAreCollections && j$.util.isUndefined(propertyNames)) {
+                propertyNames = methodNames;
+                methodNames = baseName;
+                baseName = 'unknown';
+            }
+
+            var obj = {};
+            var spiesWereSet = false;
+
+            if (j$.isArray_(methodNames)) {
+                for (var i = 0; i < methodNames.length; i++) {
+                    obj[methodNames[i]] = self.createSpy(baseName + '.' + methodNames[i]);
+                    spiesWereSet = true;
+                }
+            } else if (j$.isObject_(methodNames)) {
+                for (var key in methodNames) {
+                    if (methodNames.hasOwnProperty(key)) {
+                        obj[key] = self.createSpy(baseName + '.' + key);
+                        obj[key].and.returnValue(methodNames[key]);
+                        spiesWereSet = true;
+                    }
+                }
+            }
+
+            if (j$.isArray_(propertyNames)) {
+                for (var j = 0; j< propertyNames.length; j++) {
+                    obj[propertyNames[j]] = self.createSpy(baseName + '.' + propertyNames[j]);
```

Comment:
I think this will create spy functions on the resulting object and not spy properties. Take a look at [SpyRegistry.js](https://github.com/jasmine/jasmine/blob/516fe6875959dffde38cd512f1b19af9a862f98c/src/core/SpyRegistry.js#L68) for how property spies are currently installed.

Reply:
I'm trying to port this change on my code and managed to hit the same issue.

The spied property needs to be redefined using `Object.defineProperty` call. I believe you should be able to use `spyOnProperty` call to do this.

Afterwards, we need to expose the spy itself. Properties are not called like methods, so we can't use the same name. Contrast this with method spies: `spyObj.method` will return the spy, while `spyObj.method()` will call the spy. We want `spyObj.prop` to call the spy, and something else to return the spy.

To resolve this, we can store property spies into a dictionary and expose that. The end goal can be something like:
```
var propertySpies = { };

for (var j = 0; j< propertyNames.length; j++) {
  Object.defineProperty(obj, propertyNames[j], {
    configurable: true,
    get() { return undefined; }
  });

  propertySpies[propertyNames[j]] = spyOnProperty(obj, propertyNames[j]);
}

obj.propertySpies = propertySpies
```

And the following code should work:
```
var spiedObj = jasmine.createSpyObj("name", ["method"], ["prop"]);
spiedObj.propertySpies.prop.and.returnValue(1);
spiedObj.prop === 1; // true
```

Reply:
I don't think you need to use `spyOn` or `spyOnProperty` when building up a spy object. To set up a property spy correctly, you will have to use `Object.defineProperty`, and probably 2 spies (one for `get` and one for `set`). I would err on the side of passing the spies directly in the `defineProperty` call and not calling `spyOnProperty` afterwards unless you have to.

`spyOn` and `spyOnProperty` do some extra bookkeeping to clean up spies, and that isn't needed (or wanted) for a one-off object like that created by `createSpyObj`.